src/ScriptBuilder/Writers/SubscriptionWriter.cs

```diff
@@ -1,21 +1,41 @@
-﻿using System.IO;
+﻿using System;
+using System.IO;
 using NServiceBus.Persistence.Sql.ScriptBuilder;
 
 class SubscriptionWriter
 {
-    public static void WriteSubscriptionScript(string scriptPath, BuildSqlDialect sqlDialect)
+    public static void WriteSubscriptionScript(bool clean, bool overwrite, string scriptPath, BuildSqlDialect sqlDialect)
     {
         var createPath = Path.Combine(scriptPath, "Subscription_Create.sql");
-        File.Delete(createPath);
+        if (clean)
+        {
+            File.Delete(createPath);
+        }
+
+        CheckExistingFile(overwrite, createPath);
         using (var writer = File.CreateText(createPath))
         {
             SubscriptionScriptBuilder.BuildCreateScript(writer, sqlDialect);
         }
+
         var dropPath = Path.Combine(scriptPath, "Subscription_Drop.sql");
-        File.Delete(dropPath);
+        if (clean)
+        {
+            File.Delete(dropPath);
+        }
+
+        CheckExistingFile(overwrite, dropPath);
         using (var writer = File.CreateText(dropPath))
         {
             SubscriptionScriptBuilder.BuildDropScript(writer, sqlDialect);
         }
     }
+
+    static void CheckExistingFile(bool overwrite, string sagaFileName)
```

Comment:
Should be `subscriptionFileName` or maybe just `fileName`.

Reply:
won't be applicable after the refactoring to the base class.